third_party/move/move-vm/runtime/src/loader/function.rs

```diff
@@ -240,45 +255,47 @@ impl LazyLoadedFunction {
             .map_err(|err| err.to_partial())?;
 
         // Verify that the function argument types match the layouts used for deserialization.
-        // This is only done in paranoid mode. Since integrity of storage
-        // and guarantee of public function, this should not able to fail.
-        if module_storage
```

Comment:
what happened to this if? is it the RTT should perform checks now? I see the new comment about actual layout comparison, but we still have some code here that's irrelevant unless we do the checks? (it might be cheap, but still)

Reply:
We need to charge gas before the paranoid check otherwise paranoid checks will cost gas and cannot be turned on / off without feature gating? The trade off is that if it is off, we do extra work which is bad, I am not sure if there is an easy way around it though...

For `RTTCheck`: it is initialized to return true for "should perform checks" if paranoid flag is on. So this is the same it is we never propagated this static config to here and instead fetched the same dynamic config.